File: taskboardbackend/src/main/java/com/taskboardbackend/taskboardbackend/model/Task.java
package com.taskboardbackend.taskboardbackend.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "tasks")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String description;



    private Integer position;

    @Column(nullable = false)
    private String priority;

    @Column(name = "due_date")
    private LocalDateTime dueDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "column_id", nullable = false)
    private Columns column;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by", nullable = false)
    private User createdBy;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assignee_id", nullable = false)
    private User assigneeUser;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }


}

File: frontend/src/types/task.ts
export interface TaskRequest {
  title: string;
  description?: string;
  column: string;
  position: number;
  priority?: string;
  dueDate: Date | string;
  createdBy: string;
  assignee?: string;
}

export interface TaskResponse {
  id: string;
  title: string;
  description: string;
  column: string;
  position: number;
  priority?: string;
  dueDate: Date;
  assignee?: string;
  createdBy: string;
  createdAt: Date;
  updatedAt: Date;
}

export interface Task {
  id: string;
  title: string;
  description: string;
  position: number;
  priority?: string;
  dueDate?: Date;
  assignee?: string;
  createdBy: string;
  createdAt: Date;
  updatedAt: Date;
}

export interface MoveTaskRequest {
  targetColumnId: string;
  position?: number;
}
 